src/unix/qnx.c

```diff
@@ -0,0 +1,135 @@
+/* Copyright libuv contributors. All rights reserved.
+  *
+  * Permission is hereby granted, free of charge, to any person obtaining a copy
+  * of this software and associated documentation files (the "Software"), to
+  * deal in the Software without restriction, including without limitation the
+  * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+  * sell copies of the Software, and to permit persons to whom the Software is
+  * furnished to do so, subject to the following conditions:
+  *
+  * The above copyright notice and this permission notice shall be included in
+  * all copies or substantial portions of the Software.
+  *
+  * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+  * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+  * IN THE SOFTWARE.
+  */
+
+#include "uv.h"
+#include "internal.h"
+
+#include <string.h>
+#include <sys/process.h>
+#include <sys/neutrino.h>
+#include <sys/memmsg.h>
+#include <sys/syspage.h>
+#include <sys/procfs.h>
+
+static void
+get_mem_info(uint64_t* totalmem, uint64_t* freemem) {
+  mem_info_t  msg = { .i.type = _MEM_INFO, .i.fd = -1 };
```

Comment:
As a rule, libuv doesn't use C99 initializers. It doesn't matter much here because it's a platform-specific file but it's unidiomatic.

Also... :-)
```suggestion
  mem_info_t msg = { .i.type = _MEM_INFO, .i.fd = -1 };
```

Reply:
Fixed. Our coding standard, following MISRA, mandates designated initializers and curly braces around one-line if statement bodies, so it's a habit.
How do you deal with structures potentially changing between releases? Would you allow:

```
mem_info_t msg = { 0 }; // Make sure all fields are 0
msg.i.type = _MEM_INFO;
msg.i.fd = -1;
```

Reply:
Libuv idiom is to memset() it.

I don't know if valgrind is a thing on QNX but a secondary benefit of memset() is that it gets rid of uninitialized struct padding warnings.

Reply:
Yes, Valgrind is a thing ;-)
There should not be a problem with designated initializers, as the standard is quite clear - any field not specified is set to 0.
I've changed the code to use `memset()`.

Reply:
Oh, what I mean is the padding between fields. Say you have a struct like this:
```c
struct syscall_arg {
  uint16_t x;
  // 2 byte padding gap on most architectures
  uint32_t y;
};
```
Then valgrind has a habit of complaining if you don't zero the gap. I've sent patches in the past to fix some of those warnings but patches tend to linger in their bug tracker and releases are very infrequent anyway.

Reply:
I see what you mean, makes sense. For the record, we do not leave padding gaps in public structures, though you are correct that, at least in theory, unless you use the packed attribute you can miss something.